prometheus_hardware_exporter/collectors/redfish.py

```diff
@@ -287,21 +309,22 @@ def get_storage_controller_data(self) -> Tuple[Dict[str, int], Dict[str, List]]:
             for storage_id in storage_ids:
                 # eg: /redfish/v1/Systems/1/Storage/XYZ123
                 curr_storage_uri = (
-                    RedfishHelper._storage_root_uri(system_id, storage_name) + storage_id
+                    RedfishHelper._storage_root_uri(system_id, storage_name) + "/" + storage_id
                 )
 
                 # list of storage controllers for that storage id
-                storage_controllers_list: List[Dict] = self.redfish_obj.get(curr_storage_uri).dict[
-                    "StorageControllers"
-                ]
+                storage_controllers_list: List[Dict] = self._find_storage_controller_data(
+                    self.redfish_obj.get(curr_storage_uri).dict
+                )
                 storage_controller_count[system_id] += len(storage_controllers_list)
 
                 # picking out the required data from each storage controller in the list
                 for data in storage_controllers_list:
+                    print(data)
                     storage_controller_data_in_curr_system.append(
                         {
                             "storage_id": storage_id,
-                            "controller_id": data["MemberId"],
+                            "controller_id": data["MemberId"] if data.get("MemberId", None) else data.get("Id"),
```

Comment:
docs: A comment here as well mentioning that `Id` is what we expect if the key being used is `Controllers` in the storage data would be helpful